### ISSUE TITLE
[BUG] Pandoc installation missing in documentation build page

### ISSUE THREAD

**Issue by inclinedadarsh** [aeon-toolkit/aeon#2467]
### Describe the issue

While following the [Developer's guide for building documentation](https://www.aeon-toolkit.org/en/stable/developer_guide/documentation.html), after running the command `make html`, it gives the following error:

```
Notebook error:
PandocMissing in examples/anomaly_detection/anomaly_detection.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
make: *** [Makefile:20: html] Error 2
```

Which suggests [Pandoc](https://pandoc.org/) is missing

### Suggest a potential alternative/fix

Simply [installing pandoc](https://pandoc.org/installing.html) worked on my system.

A simple note regarding this in the [Developer's guide for building documentation](https://www.aeon-toolkit.org/en/stable/developer_guide/documentation.html) should work, but I believe there can be better solutions.

### Additional context

I would be happy to work on this issue.

**Comment by baraline** [comment]
I think a simple fix would be to add pandoc as a dependency in the developer/documentation install options in the `pyproject.toml`. 
I'm just wondering why we never had the issue before ? @MatthewMiddlehurst do we actually use this ? I'm not seeing it in the makefile or in docs.

**Comment by inclinedadarsh** [comment]
@baraline `Pandoc` is not a python dependency actually.

Sorry for not mentioning this in the original issue, but I had to fix it using `sudo apt install pandoc`.
I use Debian Bookwarm (12)

It might be just for my system, I'm not sure though.

I have mentioned the error logs in the original message, however, if you want, I can try rebuilding the docs in a different system and provide logs.

**Comment by baraline** [comment]
That would be great if you could, so we can now if it is OS related or not

**Comment by inclinedadarsh** [comment]
So I tried following the installation and documentation build steps on a fresh installation of Ubuntu (latest stable) (on wsl2) and I got the same error:
```
Notebook error:
PandocMissing in examples/anomaly_detection/anomaly_detection.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
make: *** [Makefile:20: html] Error 2
```

I ran the following commands (in order):
```bash
python3 -m venv venv
```
```bash
pip install --editable .[dev]
```
```bash
pip install --editable .[docs]
```
```bash
cd docs
```
```bash
make html
```

After doing a bit of research, it turns out `Pandoc` can be installed via `apt` as well as via `pip`
So I ran the command `pip install pandoc` and then I ran `make html` and the build was still throwing the same error.

Finally, I tried installing `pandoc` via `apt`:
```
sudo apt install pandoc
```
This resulted in a successful build, and I was able to view all the build files.

**Comment by baraline** [comment]
Thanks ! It would be nice to add this solution to the docs specifying the error and the resolution. Don't know how it will go on Windows tho, we'll document it when we encounter the issue I guess !

**Comment by shreyap2702** [comment]
I tried following the installation too, and it exited with the same error message: 
```sh
Notebook error:
PandocMissing in examples/anomaly_detection/anomaly_detection.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
make: *** [html] Error 2
```

System info: Mac M1 Air (MacOS Sonoma)

**Comment by inclinedadarsh** [comment]
Sure @baraline 

So should I proceed to add a note in the developer guide in the documentation section?

It'll be preferably in this section: https://www.aeon-toolkit.org/en/stable/developer_guide/documentation.html#documentation-build

**Comment by baraline** [comment]
@inclinedadarsh yes please, thank, not much free time for coding recently! 😄 You could refer to pandoc installation guide aswell (https://pandoc.org/installing.html)

@shreyap2702 do a ` brew install pandoc` fix the issue ?

**Comment by inclinedadarsh** [comment]
Sure. I'll submit a PR soon about it.

**Comment by inclinedadarsh** [comment]
@aeon-actions-bot assign @inclinedadarsh

**Comment by shreyap2702** [comment]
yes, `brew install pandoc` fixes the issue and produces html files

**Comment by MatthewMiddlehurst** [comment]
Never seen this before but I dev on Windows so its basically a different world 🙂. Happy to have a step or troubleshooting section which helps with this given that multiple people are having the issue.

**Comment by inclinedadarsh** [comment]
Hello @baraline & @MatthewMiddlehurst 
I have opened the PR https://github.com/aeon-toolkit/aeon/pull/2489 for this.

Please review it and let me know if any other changes are required.
